pkg/controller/component/synthesize_component.go

```diff
@@ -255,6 +257,32 @@ func buildComp2CompDefs(cluster *appsv1alpha1.Cluster, clusterCompSpec *appsv1al
 	return mapping
 }
 
+// buildLabels builds labels for synthesizedComponent.
+func buildLabels(compDef *appsv1alpha1.ComponentDefinition, comp *appsv1alpha1.Component, synthesizeComp *SynthesizedComponent) {
+	ReplaceKBEnvPlaceholderTokens := func(clusterName, uid, componentName, strToReplace string) string {
+		builtInEnvMap := GetReplacementMapForBuiltInEnv(clusterName, uid, componentName)
+		return ReplaceNamedVars(builtInEnvMap, strToReplace, -1, true)
+	}
+
+	labels := make(map[string]string)
+	if compDef.Labels == nil && comp.Labels == nil {
+		return
+	}
+
+	for k, v := range compDef.Spec.Labels {
+		resolveKey := ReplaceKBEnvPlaceholderTokens(synthesizeComp.ClusterName, synthesizeComp.ClusterUID, synthesizeComp.Name, k)
+		resolveValue := ReplaceKBEnvPlaceholderTokens(synthesizeComp.ClusterName, synthesizeComp.ClusterUID, synthesizeComp.Name, v)
+		labels[resolveKey] = resolveValue
+	}
+
+	// override labels from component
+	for k, v := range comp.Labels {
```

Comment:
intend to sync all component labels to workloads?

Reply:
> intend to sync all component labels to workloads?

yes, there has been a requirement received before where pods are intended to inherit labels from both the cluster and component. These labels could be either externally defined by the user or internally defined by KubeBlocks. However, there is a potential risk that modifications to the pod template labels could lead to restart issues.

Reply:
> > intend to sync all component labels to workloads?
> 
> yes, there has been a requirement received before where pods are intended to inherit labels from both the cluster and component. These labels could be either externally defined by the user or internally defined by KubeBlocks. However, there is a potential risk that modifications to the pod template labels could lead to restart issues.

reverted